### PR TITLE
Preliminary unit tests for functional package

### PR DESCRIPTION
**Pull request by JacobGabrielson**
Issue #, if available: n/a

Description of changes: preliminary unit tests for `functional` pkg

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ellistarn — DISMISSED** [review]
I'm not sure I'm sold on this heavy usage of the DLS (e.g. When, Specify, etc).

I'd probably use the minimum necessary, e.g.
```
Describe("Suite", func() {
  It("should be a test case", func() {
  })
})
```
I could see using additional context blocks as necessary to setup variables at different scopes, but if no setup is being done within a block, we might want to collapse it. WDYT?

**Comment by JacobGabrielson** [comment]
> I'm not sure I'm sold on this heavy usage of the DLS (e.g. When, Specify, etc).
> 
> I'd probably use the minimum necessary, e.g.
> 
> ```
> Describe("Suite", func() {
>   It("should be a test case", func() {
>   })
> })
> ```
> 
> I could see using additional context blocks as necessary to setup variables at different scopes, but if no setup is being done within a block, we might want to collapse it. WDYT?

Makes sense - cut down on levels of indirection, LMKWYT?

File: pkg/utils/functional/suite_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+)
+
+func TestFunctional(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Functional Suite")
+}
+
+var _ = Describe("Functional", func() {
+	Context("UnionStringMaps", func() {
+		empty := make(map[string]string)
```
Comment (ellistarn) [inline pkg/utils/functional/suite_test.go:17]: I'd probably just define the maps inline. [string]string{} is only a few extra characters and more explicit.

Comment (ellistarn) [inline pkg/utils/functional/suite_test.go:17]: It would be more logical to me if this were initialized as a sibling to the other canonical maps.

**Review by ellistarn — APPROVED** [review]
One optional nit.

**Review by ellistarn — DISMISSED** [review]

File: pkg/utils/functional/suite_test.go (hunk 1 above)
Comment (JacobGabrielson) [reply pkg/utils/functional/suite_test.go]: Fixing

**Comment by JacobGabrielson** [comment]
> One optional nit.

fix'd

**Review by ellistarn — APPROVED** [review]

**Pull request merged**